RInAppMessaging/Classes/Models/UserInfoProvider.swift

```diff
@@ -0,0 +1,35 @@
+/**
+ * Interface which client app should implement in order for InAppMessaging SDK to get information
+ * when needed.
+ */
+@objc public protocol UserInfoProvider {
+
+    /**
+     * Only return RAE token if user is logged in. Else return null.
+     *
+     * @return String of RAE token.
+     */
+    var provideRaeToken: String? { get }
+
+    /**
+     * Only return user ID used when logging if user is logged in in the current session.
+     *
+     * @return String of the user ID.
+     */
+    var provideUserId: String? { get }
+
+    /**
+     * Only return Rakuten ID used in the current session.
+     *
+     * @return String of the Rakuten ID.
+     */
+    var provideRakutenId: String? { get }
+}
+
+extension UserInfoProvider {
+    // value used to check UserInfoProvider for equality
+    // swiftlint:disable:next legacy_hashing
+    var hashValue: Int {
```

Comment:
>Or is it about Obj-C compatibility?

```swift
/// Conforms to NSObject and exposed with objc tag to make it work with Obj-c projects.
@objc public class RInAppMessaging: NSObject {
    @objc public static func registerPreference(_ preference: UserInfoProvider?) {
```

Had trouble implementing `Equatable` because of this. Maybe we handle equality with `isEqual(_:)->Bool` off of the `NSObjectProtocol` if we want to keep `@objc`